### PR TITLE
catch2 2.x: fix package_type

### PR DESCRIPTION
**Pull request by SpaceIm**
catch2 2.x is header only, or static if with_main=True, there is no shared option.
`package_type = "library"` fails with conan v2 client when there is no shared option.

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by valgur** [comment]
Would `package_type = "library-static"` be a better alternative, perhaps?

**Comment by SpaceIm** [comment]
Why does conan v2 pipeline still report `system is under maintenance`?

**Comment by SpaceIm** [comment]
> Would `package_type = "library-static"` be a better alternative, perhaps?

catch2 is header-only by default, and static if with_main=True.

I guess it should be something like this:

```python
def configure(self):
    self.package_type = "static-library" if self.options.with_main else "header-library"
```

File: recipes/catch2/2.x.x/conanfile.py [hunk 1]
```diff
@@ -42,6 +41,7 @@ def config_options(self):
             del self.options.fPIC
 
     def configure(self):
+        self.package_type = "static-library" if self.options.with_main else "header-library"
```
Comment (prince-chrismc) [inline recipes/catch2/2.x.x/conanfile.py:44]: It can not be dynamically set AFAIK -- these should be different recipes to have this seperation

Comment (prince-chrismc) [reply recipes/catch2/2.x.x/conanfile.py:44]:
```suggestion
```

Comment (SpaceIm) [reply recipes/catch2/2.x.x/conanfile.py:44]: It can obviously since CI passes, and catch2 2.x can't be split in 2 different recipes.

Comment (prince-chrismc) [reply recipes/catch2/2.x.x/conanfile.py]: Sorry if I was unclear... but i think we are not on the same page here. Let me try again

CCI does not test the consumer graph combinations 😕 So I am not sure why that is relevant here.
The new graph was original designed using only options which is why that is the preferred way of doing it. (i.e omit the package type completely is supported)

I see you do not like the UX but the code you are adding is untested in the client to my knowledge and should not be done at this time. It's unclear what consequences it can have since the graph would change after being loaded and might have edge cases.

**Review by prince-chrismc — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]
